### ISSUE TITLE
vim opens multiple files with garbled GETTEXT messages

### ISSUE THREAD

**Issue by lsq** [vim/vim#15752]
### Steps to reproduce

Install vim from vim-win32-installer, then open MINGW64 zsh in windows terminal or open cmd.exe/powershell; command
```
vim.exe main.c os_win32.c
杩樻湁 2 杓枃浠 chop 瓑寰呯紪杈
```
The prompt message is garbled; if `chcp 65001` is used before the above command, `cmd.exe` and `mingw64 zsh.exe` are normal (the `power shell` environment is still garbled, and the input/output encoding is set to utf-8 and is normal)
How to set up normal output of messages that does not depend on specific environment

https://github.com/user-attachments/assets/7701a124-b1c1-4fbf-8e15-9c2328716d64

### Expected behaviour

A normal message should say that `2 个文件等待编辑`

### Version of Vim

9.1.X

### Environment

OS: Windows 10 x86_64
Shell: zsh 5.9（mingw64) / cmd.exe /power shell
windows terminal
`:lang` output
![image](https://github.com/user-attachments/assets/f9c4a3df-19fb-4e1e-8099-6a3cacb86e9d)

**Comment by chrisbra** [comment]
Hm, I did not see the error message from your video. Does it happen with `vim --clean`? Is there a reason, you do not run in UTF-8 codepage by default? I think this is now the default for the new Windows terminal anyhow.

**Comment by lsq** [comment]
@chrisbra hey，after a week of testing, I found the reason: setlocale setting code page (cp936) is inconsistent with bind_textdomain_codeset (VIMPACKAGE, 'utf-8'). Before terminal output, the utf-8 encoding is recognized as the encoding set by setlocale (p936), so it appears Garbled code, can this be fixed?

**Comment by chrisbra** [comment]
> is inconsistent with bind_textdomain_codeset (VIMPACKAGE, 'utf-8')

Where do you have set? And isn't your Vim running with utf-8 encoding?

**Comment by lsq** [comment]
> > is inconsistent with bind_textdomain_codeset (VIMPACKAGE, 'utf-8')
> 
> Where do you have set? And isn't your Vim running with utf-8 encoding?

In `mbyte.c`,  I set bind_textdomain_codeset (VIMPACKAGE, 'cp936) replacing `enc_utf8 ? "utf-8" : (char *)p_enc`，then the message normal.

**Comment by chrisbra** [comment]
well then try running vim with `:set encoding=cp936`. Then `p_enc` should be `cp936`. But I think you should rather convert to utf-8 instead.

**Comment by lsq** [comment]
You are right, with setting `:set encoding=cp936`,  not only the message is garbled,  but also the text file's content. And `p_enc` is still `utf-8` , not `cp936`.
